apps/history/declarativehistorymodel.cpp

```diff
@@ -28,6 +31,7 @@ QHash<int, QByteArray> DeclarativeHistoryModel::roleNames() const
     roles[UrlRole] = "url";
     roles[TitleRole] = "title";
     roles[DateRole] = "date";
+    roles[FaviconRole] = "favicon";
     return roles;
 }
 
```

Comment:
Please excuse the odd comment placement (gitlab only lets you put comments near lines that have changed).

It would probably make sense to add a call to `FaviconManager::clear("history")` in the `DeclarativeHistoryModel::clear()` method, so that the icons get cleared from disc when the history is cleared.

Reply:
done